events_frontend/src/components/CreateEvent.js

```diff
@@ -172,47 +172,25 @@ class CreateEvent extends Component {
 
   uploadImage(callback) {
     const file = this.state.image;
-    let xhr = new XMLHttpRequest();
-    xhr.open(
-      "GET",
-      "/api/sign_s3/?file_name=" + file.name + "&file_type=" + file.type
-    );
-    xhr.onreadystatechange = function() {
-      if (xhr.readyState === 4) {
-        if (xhr.status === 200) {
-          const response = JSON.parse(xhr.responseText);
-          xhr = new XMLHttpRequest();
-          xhr.open("POST", response.data.url);
-
-          let postData = new FormData();
-          for (let key in response.data.fields) {
-            postData.append(key, response.data.fields[key]);
-          }
-          postData.append("file", file);
-
-          xhr.onreadystatechange = function() {
-            if (xhr.readyState === 4) {
-              if (xhr.status === 200 || xhr.status === 204) {
-                console.log("File uploaded!");
-                callback(
-                  response.url
-                    .split("/")
-                    .slice(3)
-                    .join("/")
-                );
-              } else {
-                alert("Could not upload file.");
-              }
-            }
-          };
-          xhr.send(postData);
-          //self.uploadFile(file, response.data, response.url);
-        } else {
-          alert("Could not get signed URL.");
-        }
-      }
-    };
-    xhr.send();
+    const fileType = file.type.replace("image/", "");
+
+
+    if (fileType.length == file.type.length) {
+      alert("Could not upload image. Please use a different file type");
```

Comment:
return here

Reply:
This is strange, this does not look like the code on my end. We switched to axios/promise-based flow (we also called a different endpoint.) This is what it looks like on my end:

![Screenshot (1413)](**https://user-images.githubusercontent.com/18223523/70965082-6f9f1c80-205c-11ea-8990-c850e14f9086.png**)